File: JavaScript/01.JavaScript-Basic-February-2021/09.whileLoop-Lab/08.graduationPt2.js
function graduation(input) {
    let index = 0;
    let name = input[index];
    index++;
    let sum = 0;
    let poorGrade = 0;

    while (index <= 12) {
        if (poorGrade === 2) {
            break;
        }
        let tempGrade = Number(input[index]);
        index++
        if (tempGrade >= 4) {
            sum += tempGrade;
        } else {
            poorGrade++
        }

    }

    if (poorGrade < 2) {
        console.log(`${name} graduated. Average grade: ${(sum / (index - 1)).toFixed(2)}`);
    } else {
        console.log(`${name} has been excluded at ${(index - poorGrade)} grade`);
    }

}

graduation(["Gosho", "5", "5.5", "6", "5.43", "5.5", "6", "5.55", "5", "6", "6", "5.43", "5"]);
graduation(["Mimi", "5", "6", "5", "6", "5", "6", "6", "2", "3"]);

File: Python/01.Python-Basic-April-2019/08.while_loop_lab/08.graduation_pt_2.py
name = input()

level = 1
sum_grades = 0
bad_grades_counter = 0
while level <= 12:
    if bad_grades_counter == 2:
        print(f'{name} has been excluded at {level} grade')
        break

    grade = float(input())
    if grade < 4.00:
        bad_grades_counter += 1
    else:
        level += 1
        sum_grades += grade
if bad_grades_counter < 2:
    print(f'{name} graduated. Average grade: {(sum_grades / 12):.2f}')
